tv_apps/smart-home/js/home.js

```diff
@@ -779,19 +780,23 @@
 
     updateClock: function() {
       var now = new Date();
-      var _ = navigator.mozL10n.get;
-      var use12Hour = window.navigator.mozHour12;
 
-      var f = new navigator.mozL10n.DateTimeFormat();
+      var f = mozIntl.DateTimeFormat(navigator.languages, {
+        hour: 'numeric',
+        minute: 'numeric',
+        dayperiod: false
+      });
 
-      var timeFormat = use12Hour ? _('shortTimeFormat12') :
-                                   _('shortTimeFormat24');
-      // remove AM/PM and we use our owned style to show it.
-      timeFormat = timeFormat.replace('%p', '').trim();
-      var formatted = f.localeFormat(now, timeFormat);
+      var use12Hour = f.resolvedOptions().hour12;
 
-      this.timeElem.innerHTML = formatted;
-      this.timeElem.dataset.ampm = use12Hour ? f.localeFormat(now, '%p') : '';
+      this.timeElem.innerHTML = f.format(now);
+      if (use12Hour) {
+        var amPmFmt = mozIntl.DateTimeFormat(navigator.languages, {
+          hour12: true,
+          dayperiod: true,
+        });
+        this.timeElem.dataset.ampm = use12Hour ? amPmFmt.format(now) : '';
+      }
```

Comment:
I tested it on B2G-Desktop with different timezones (Shenghai, Warsaw, Los Angeles) and it worked properly. I rebased the patch on top of master. Can you retest pls?